src/Kubernetes/KubernetesApiExtensions.cs

```diff
@@ -503,12 +503,46 @@ public static V1Job WithBillingId(this V1Job job, string billingId)
             });
         }
 
+        /// <summary>
+        /// Adds a policy failure action and exit codes to the job.
+        /// </summary>
+        /// <param name="job">The job object to modify.</param>
+        /// <param name="actions">A list with all action names and corresponding exit code number.</param>
+        /// <returns>The Kubernetes Job object with added pod failure policy rules.</returns>>
+        public static V1Job WithPodPolicyFailureExitCodes(this V1Job job, Dictionary<string, List<int>> actions)
+        {
+            job.Spec ??= new V1JobSpec();
+            job.Spec.PodFailurePolicy ??= new V1PodFailurePolicy();
+
+            var newRules = actions.Select(action => new V1PodFailurePolicyRule
+            {
+                Action = action.Key,
+                OnExitCodes = new V1PodFailurePolicyOnExitCodesRequirement
+                {
+                    Values = action.Value.Distinct().ToList()
+                }
+            }).ToList();
+
+
+            if (job.Spec.PodFailurePolicy.Rules != null)
+            {
+                job.Spec.PodFailurePolicy.Rules = job.Spec.PodFailurePolicy.Rules.Concat(newRules).ToList();
```

Comment:
I think you should merge rules rather than concatenating, since you can have overlapping actions

Reply:
I've updated it now in [34b73fc](https://github.com/SneaksAndData/esd-services-sdk/pull/75/commits/34b73fc686d9b20380c8d0b38176beec2570b4fe) as we discussed.
Thank you for the input 🙂